File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0;
        int j = 0;
        while(i < s.length() && j < t.length()) {
            if(s[i] == t[j]) {
                i++;
            }
            j++;
        }
        return i == s.length();
    }
};

File: 0392-is-subsequence/0392-is-subsequence.rs
impl Solution {
    pub fn is_subsequence(s: String, t: String) -> bool {
        let mut s_iter = s.chars();
        let mut t_iter = t.chars();
        let mut s_char = s_iter.next();

        while let Some(t_char) = t_iter.next() {
            if Some(t_char) == s_char {
                s_char = s_iter.next();
            }
        }
        s_char.is_none()
    }    
}